django/cantusdb_project/main_app/views/chant.py

```diff
@@ -103,27 +103,18 @@ def get_context_data(self, **kwargs):
         ).prefetch_related("feast")
 
         def get_chants_with_feasts(chants_in_folio):
-            feast_ids = []
+            feasts_chants = []
             for chant in chants_in_folio.order_by("sequence_number"):
-                if chant.feast:
-                    feast_ids.append(chant.feast.id)
-            # remove duplicate feast ids and preserve the order
-            feast_ids = list(dict.fromkeys(feast_ids))
-
-            feasts = []
-            for feast_id in feast_ids:
-                feasts.append(Feast.objects.get(id=feast_id))
-            # feasts = Feast.objects.filter(id__in=feast_ids) # this loses the order
-            chants_in_feast = []
-            for feast in feasts:
-                chants = chants_in_folio.filter(feast=feast).order_by("sequence_number")
-                chants_in_feast.append(chants)
-            feasts_zip = zip(feasts, chants_in_feast)
-            return feasts_zip
-
-        context["feasts_current_folio"] = list(
-            get_chants_with_feasts(chants_current_folio)
-        )
+                if chant.feast and (not feasts_chants or chant.feast.id != feasts_chants[-1][0]):
```

Comment:
I'm a bit confused about what this does. Can you add some more comments in the code?